doc.go

```diff
@@ -0,0 +1,63 @@
+/*
+Package oidc implements OpenID Connect client logic for the golang.org/x/oauth2 package.
+
+	provider, err := oidc.NewProvider(ctx, "https://accounts.exmaple.com")
+	if err != nil {
+		return err
+	}
+
+	// Configure an OpenID Connect aware OAuth2 client.
```

Comment:
All that is needed to make an OAuth2 client OpenID Connect aware is providing the right endpoint and scopes right? Just checkin